internal/hmyapi/apiv1/harmony.go

```diff
@@ -70,6 +71,11 @@ func (s *PublicHarmonyAPI) GetNodeMetadata() NodeMetadata {
 		blockEpoch = &b
 	}
 
+	//Uses GetShardingStructure to expose endpoint URL
+	numShard := shard.Schedule.InstanceForEpoch(header.Epoch()).NumShards()
+	shardStructure := shard.Schedule.GetShardingStructure(int(numShard), int(s.b.GetShardID()))
+	endpointURL := shardStructure[int(s.b.GetShardID())]["http"].(string)
```

Comment:
http or https?

Reply:
http, it's a field in the getShardingStructure response.
https://github.com/harmony-one/harmony/blob/c6b27393154c4c776b4435eed25b05a34e6dd17f/internal/configs/sharding/shardingconfig.go#L77-L88